time-series.ipynb.py

```diff
@@ -0,0 +1,143 @@
+!pip install pytorch_forecasting
+
+import warnings
+import numpy as np
+import pandas as pd
+import copy
+from pathlib import Path
+import torch
+import torch.nn as nn
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import EarlyStopping, LearningRateMonitor
+from pytorch_lightning.loggers import TensorBoardLogger
+
+import pytorch_forecasting
+from pytorch_forecasting import Baseline, TemporalFusionTransformer, TimeSeriesDataSet
+from pytorch_forecasting.data import GroupNormalizer
+from pytorch_forecasting.metrics import SMAPE, PoissonLoss, QuantileLoss
+from pytorch_forecasting.models.temporal_fusion_transformer.tuning import optimize_hyperparameters
+
+
+
+df_train = pd.read_csv('/kaggle/input/competitive-data-science-predict-future-sales/sales_train.csv' ,parse_dates=['date'])
+df_shops = pd.read_csv('/kaggle/input/competitive-data-science-predict-future-sales/shops.csv')
+df_items = pd.read_csv('/kaggle/input/competitive-data-science-predict-future-sales/items.csv')
+df_item_categories = pd.read_csv('/kaggle/input/competitive-data-science-predict-future-sales/item_categories.csv')
+df_train['date'] = pd.to_datetime(df_train['date'], errors='coerce')
+df_train.columns
+Index(['date', 'date_block_num', 'shop_id', 'item_id', 'item_price',
+       'item_cnt_day'],
+      dtype='object')
+
+# change the item count per day to item count per month by using group
+df_train = df_train.groupby(["item_id","shop_id","date_block_num"]).sum().reset_index()
+df_train = df_train.rename(index=str, columns = {"item_cnt_day":"item_cnt_month"})
+df_train = df_train[["item_id","shop_id","date_block_num","item_cnt_month"]]
+
+
+df_train['date_block_num'].describe()
+
+
+max_prediction_length = 1
+max_encoder_length = 27
+training_cutoff = df_train['date_block_num'].max() - max_prediction_length
+
+training = TimeSeriesDataSet(

```

Comment:
min/max scaler is super decent to use in this case,  before any training..
I would consider:

from sklearn.preprocessing import MinMaxScaler

scaler = MinMaxScaler(feature_range=(-1, 1))
df_train_normalized = scaler.fit_transform(df_train .reshape(-1, 1))